physics/src/modules/include/NextsimPhysics.hpp

```diff
@@ -0,0 +1,272 @@
+/*!
+ * @file NextsimPhysics.hpp
+ * @date Sep 9, 2021
+ * @author Tim Spain <timothy.spain@nersc.no>
+ */
+
+#ifndef SRC_INCLUDE_NEXTSIMPHYSICS_HPP
+#define SRC_INCLUDE_NEXTSIMPHYSICS_HPP
+#include <memory>
+
+#include "include/BaseElementData.hpp"
+#include "include/Configured.hpp"
+#include "include/IPhysics1d.hpp"
+#include "include/constants.hpp"
+
+namespace Nextsim {
+
+class PrognosticData;
+class PhysicsData;
+class ExternalData;
+class UnusedData;
+
+class IIceAlbedo;
+class IThermodynamics;
+class IIceOceanHeatFlux;
+class IConcentrationModel;
+
+template <class Phys> class ElementData;
+
+class NextsimPhysics;
+
+class NextsimPhysics : public BaseElementData,
+                       public Configured<NextsimPhysics>,
+                       public IPhysics1d {
+public:
+    NextsimPhysics();
+
+    void configure() override;
+    enum {
+        DRAGOCEANQ_KEY,
+        DRAGOCEANT_KEY,
+        DRAGICET_KEY,
+        OCEANALBEDO_KEY,
+        I0_KEY,
+        MINC_KEY,
+        MINH_KEY,
+    };
+
+    void calculate(const PrognosticData&, const ExternalData&, PhysicsData&) override;
+
+    //! Calculate the new ice formed this timestep on open water
+    void newIceFormation(const PrognosticData& prog, const ExternalData& exter, PhysicsData& phys);
+    //! The thickness of newly created ice in the current timestep
+    inline double newIce() const { return m_newice; };
+
+    //! Total ice-atmosphere heat flux [W m⁻²]
+    inline double QIceAtmosphere() const { return m_Qia; };
+    //! Ice to ocean heat flux [W m⁻²]
+    inline double& QIceOceanHeat() { return m_Qio; };
```

Comment:
Is this better than declaring ``m_Qio`` public? You do this quite a lot, but to me the assignment ``phys.QIceOceanHeat() = 0.`` (for example) is then a bit counterintuitive, because you're assigning something to the result of a function call. But maybe that's just my Fortran background?